src/Language/Visitor.php

```diff
@@ -224,7 +225,9 @@ public static function visit(object $root, array $visitor, ?array $keyMap = null
                             ++$editOffset;
                         } else {
                             if ($node instanceof NodeList || \is_array($node)) {
-                                $node[$editKey] = $editValue;
+                                if ($editValue !== false) {
```

Comment:
I think we should throw if `editValue` is not a `Node` here

Reply:
See the commit, that breaks tests since it happens very often (most of visitors return void, which is coerced to false).

Reply:
> coerced to false

`void` will become `null`.

Reply:
You're right, I got lost in callbacks.

But if I throw, everything kind of breaks apart. `false` seems to be a valid value, just not something that needs to be added to the NodeList

Reply:
Found and fixed the culprit.

Reply:
nice